### PR TITLE
feat(logfile): add automatic logfile recorder/player for CLIs

### PR DESCRIPTION
**Pull request by jaredallard**
## What this PR does / why we need it

This PR adds an automatic cli logfile generator as well as player for said log files

I also removed the current tracing and logging code in the current CLI helpers, as well as better organized it. The logfile work is contained to `pkg/cli/logfile`

## Jira ID

[DT-0]

### CODE REVIEW

File: pkg/cli/cli.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+
+// Description: See package comment
```
Comment (jkinkead) [inline pkg/cli/cli.go:3]: This is for the file contents, isn't it? This package has multiple files in it - shouldn't this describe the contents of this file?

File: pkg/cli/shutdown.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+		// Go sets panic exit codes to 2
+		(*exitCode) = 2
+	}
+}
+
+// setupExitHandler sets up an exit handler that will call os.Exit() with
+// the set exit code, ensuring that all log/trace data is flushed.
+func setupExitHandler() (exitCode *int, exit func()) {
+	exitCode = intPtr(0)
```
Comment (jkinkead) [inline pkg/cli/shutdown.go:43]: Can't you just use a local and get a pointer to that?
```suggestion
	exitCodeInt := 0
	exitCode := &exitCodeInt
```

**Review by jkinkead — COMMENTED** [review]
Two short comments, may review fully later ...

File: pkg/cli/cli.go (hunk 1 above)
Comment (jaredallard) [reply pkg/cli/cli.go:3]: It is, but the package comment describes what's in this file. I can duplicate it if you'd prefer

File: pkg/cli/shutdown.go (hunk 2 above)
Comment (jaredallard) [reply pkg/cli/shutdown.go:43]: Yeah, this is just copied code. Will adjust since i need to remove that function as well

**Review by artemchernyak-outreach — APPROVED** [review]

File: pkg/cli/cli.go (hunk 1 above)
Comment (jkinkead) [reply pkg/cli/cli.go:3]: Shouldn't you fix the package comment, then?

Comment (jaredallard) [reply pkg/cli/cli.go:3]: Yeah, I guess it was slightly messed up. Should be better now. LMK

**Comment by jaredallard** [comment]
Closing until this can be taken over again :(

**Pull request closed without merging**